common/table.js

```diff
@@ -1047,8 +1117,31 @@
             }
 
             scope.toggleFacetPanel = function () {
-                var action = (scope.vm.config.facetPanelOpen ? logActions.recordsetFacetClose : logActions.recordsetFacetOpen );
-                logService.logAction(action, logActions.clientAction);
+                var action;
+                var panelOpen = scope.vm.config.facetPanelOpen;
+                // facet panel can only be toggled in main recordset, p&b add, and fk add/edit
+                switch (scope.vm.config.displayMode) {
+                    case recordsetDisplayModes.fullscreen:
+                        action = (panelOpen ? logActions.recordsetFacetClose : logActions.recordsetFacetOpen );
+                        break;
+                    case recordsetDisplayModes.addPureBinaryPopup:
+                        action = (panelOpen ? logActions.recordPBClose : logActions.recordPBOpen );
+                        break;
+                    case recordsetDisplayModes.foreignKeyPopupCreate:
+                    case recordsetDisplayModes.foreignKeyPopupEdit:
+                        action = (panelOpen ? logActions.recordeditFKClose : logActions.recordeditFKOpen );
+                        break;
+                    default:
+                        break;
+                }
+
+                var toggleFacetPanelHeader = {
```

Comment:
After reading the rest of the code, I realized in some places you're logging more than just the `action`, `catalog`, `schema_table`.

You could add `reference` as an optional parameter to your function, so if it's passed you're just going to attach `catalog` and `schema_table` to the passed object.

```javascript
function logClientAction(headerObj, reference) {
  if (!cc.logClientActions) return;	

  if (reference) {
    headerObj.catalog = reference.defaultLogInfo.catalog;
    headerObj.schema_table = reference.defaultLogInfo.schema_table;
  }

  // the rest of function
}
```

Although the more I think about it, I feel like this might be unnecessary. My first reaction after seeing all these repetitions was that maybe we could refactor it, but that might not be necessary really.

Reply:
I also made this change because of how messy the code was the way I originally wrote it.